network/manager_test.go

```diff
@@ -0,0 +1,234 @@
+					err := nm.save()
+					Expect(err).NotTo(HaveOccurred())
+					Expect(nm.TimeStamp).To(Equal(time.Time{}))
+				})
+			})
+			Context("When store.Write error", func() {
+				It("Should raise error", func(){
+					nm := &networkManager{
+						store: &testutils.KeyValueStoreMock{
+							WriteError: errors.New("error for test"),
+						},
+					}
+					err := nm.save()
+					Expect(err).To(HaveOccurred())
+					Expect(nm.TimeStamp).NotTo(Equal(time.Time{}))})
+			})
+		})
+
+		Describe("Test GetNumberOfEndpoints", func() {
+
+			Context("When ExternalInterfaces is nil", func() {
+				It("Should return 0", func() {
+					nm := &networkManager{}
+					num := nm.GetNumberOfEndpoints("", "")
+					Expect(num).To(Equal(0))
+				})
+			})
+
+			Context("When extIf not found", func() {
+				It("Should return 0", func() {
+					nm := &networkManager{
+						ExternalInterfaces: map[string]*externalInterface{},
+					}
+					num := nm.GetNumberOfEndpoints("eth0", "")
+					Expect(num).To(Equal(0))
+				})
+			})
+
+			Context("When Networks is nil", func() {
+				It("Should return 0", func() {
+					ifName := "eth0"
+					nm := &networkManager{
+						ExternalInterfaces: map[string]*externalInterface{},
+					}
+					nm.ExternalInterfaces[ifName] = &externalInterface{}
+					num := nm.GetNumberOfEndpoints(ifName, "")
+					Expect(num).To(Equal(0))
+				})
+			})
+
+			Context("When network not found", func() {
+				It("Should return 0", func() {
+					ifName := "eth0"
+					nm := &networkManager{
+						ExternalInterfaces: map[string]*externalInterface{},
+					}
+					nm.ExternalInterfaces[ifName] = &externalInterface{
+						Networks: map[string]*network{},
+					}
+					num := nm.GetNumberOfEndpoints(ifName, "nwId")
+					Expect(num).To(Equal(0))
+				})
+			})
+
+			Context("When endpoints is nil", func() {
+				It("Should return 0", func() {
+					ifName := "eth0"
+					nwId := "nwId"
+					nm := &networkManager{
+						ExternalInterfaces: map[string]*externalInterface{},
+					}
+					nm.ExternalInterfaces[ifName] = &externalInterface{
+						Networks: map[string]*network{},
+					}
+					nm.ExternalInterfaces[ifName].Networks[nwId] = &network{}
+					num := nm.GetNumberOfEndpoints(ifName, nwId)
+					Expect(num).To(Equal(0))
+				})
+			})
+
+			Context("When endpoints is found", func() {
+				It("Should return the length of endpoints", func() {
+					ifName := "eth0"
+					nwId := "nwId"
+					nm := &networkManager{
+						ExternalInterfaces: map[string]*externalInterface{},
+					}
+					nm.ExternalInterfaces[ifName] = &externalInterface{
+						Networks: map[string]*network{},
+					}
+					nm.ExternalInterfaces[ifName].Networks[nwId] = &network{
+						Endpoints: map[string]*endpoint{
+							"ep1":&endpoint{},
+							"ep2":&endpoint{},
+							"ep3":&endpoint{},
+						},
+					}
+					num := nm.GetNumberOfEndpoints(ifName, nwId)
+					Expect(num).To(Equal(3))
+				})
+			})
+
+			Context("When ifName is empty", func() {
```

Comment:
couldnt understand this? what you mean by ifName is empty? Can you add a proper description of this test case

Reply:
"When ifName is not specified"

Reply:
can you update description saying "When ifName not specifed in GetNumberofEndpoints"